apps/meteor/app/livechat/server/lib/QueueManager.js

```diff
@@ -73,7 +73,13 @@ export const QueueManager = {
 		await queueInquiry(room, inquiry, agent);
 		logger.debug(`Inquiry ${inquiry._id} queued`);
 
-		return LivechatRooms.findOneById(rid);
+		const newRoom = await LivechatRooms.findOneById(rid);
+		if (!newRoom) {
+			logger.debug(`Room with id ${rid} not found`);
+			throw new Error('room-not-found');
```

Comment:
what happens to this error thrown? is it caught somewhere?

Reply:
Well we're not explicitly catching this error, rather it's being handled as a generic error. Which means the calling service/lib will be responsible for handling this error. So if it's an API call, then this error would be returned within the error message of the api response. On the other hand, if this is an App's engine call, then this error would be thrown to app and they'd be responsible for handling it appropriately based on their use-case.
Additionally I don't really see this error being thrown since this room was just created by the system just a few lines before this code.